products/appengine/api.yaml

```diff
@@ -457,8 +458,93 @@ objects:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'standardSchedulerSettings'
+            description: |
+              Scheduler settings for standard environment.
+            properties:
+              - !ruby/object:Api::Type::Double
+                name: 'targetCpuUtilization'
+                description: |
+                  Target CPU utilization ratio to maintain when scaling. Should be a value in the range [0.50, 0.95], zero, or a negative value.
+              - !ruby/object:Api::Type::Double
+                name: 'targetThroughputUtilization'
+                description: |
+                  Target throughput utilization ratio to maintain when scaling. Should be a value in the range [0.50, 0.95], zero, or a negative value.
+              - !ruby/object:Api::Type::Integer
+                name: 'minInstances'
+                description: |
+                  Minimum number of instances to run for this version. Set to zero to disable minInstances configuration.
+              - !ruby/object:Api::Type::Integer
+                name: 'maxInstances'
+                description: |
+                  Maximum number of instances to run for this version. Set to zero to disable maxInstances configuration.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'basicScaling'
+        description: |
+          Basic scaling creates instances when your application receives requests. Each instance will be shut down when the application becomes idle. Basic scaling is ideal for work that is intermittent or driven by user activity.
+        conflicts:
+          - automaticScaling
+          - manualScaling
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'idleTimeout'
+            default_value: 900s
+            description: |
+              Duration of time after the last request that an instance must wait before the instance is shut down.
+              A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s". Defaults to 900s.
+          - !ruby/object:Api::Type::Integer
+            name: 'maxInstances'
+            required: true
+            description: |
+              Maximum number of instances to create for this version. Must be in the range [1.0, 200.0].
+      - !ruby/object:Api::Type::NestedObject
+        name: 'manualScaling'
+        description: |
+          A service with manual scaling runs continuously, allowing you to perform complex initialization and rely on the state of its memory over time.
+        conflicts:
+          - automaticScaling
+          - basicScaling
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'instances'
+            required: true
+            description: |
+              Number of instances to assign to the service at the start. This number can later be altered by using the Modules API set_num_instances() function.
```

Comment:
Does it make sense to be able to set this via terraform if it is explicitly supposed to be updated later via a different interface?

I'm thinking of the case where someone sets a value via terraform, then updates it via another interface. Generally the API will start returning the new value, causing a diff as terraform tries to reset it to the original value. Maybe we just want to be able to set this and require users to ignore it via lifecycle ignore_changes?

Do you have an idea of how this will handle changes to this field?

Reply:
The not about updating using the Modules API came from the REST API docs. I looked into this and this only applies to the python 2 appEngine (which is now deprecated). I clarified that in the updated description.
For all other runtimes you would just have to use the AppEngine Admin API to update the scaling properties. 
However, I do see why there are use cases where it makes sense to adjust the manual-scaling properties at runtime. I've added a note to the description explaining that users that intend to adjust this at runtime (outside of terraform) should specify this in `ignore_changes`.